syntax = "proto3";

package maarpc;

import "types.proto";

message SetGlobalOptionRequest {
  oneof option {
    string logging = 1;
    bool debug_mode = 2;
  }
}

service Utility {
  rpc version(EmptyRequest) returns (StringResponse);
  rpc set_global_option(SetGlobalOptionRequest) returns (EmptyResponse);

  rpc acquire_id(EmptyRequest) returns (IdResponse);

  rpc register_callback(IdRequest) returns (stream Callback);
  rpc unregister_callback(IdRequest) returns (EmptyResponse);
}
